examples/platform/k32w/app/support/Memconfig.cpp

```diff
@@ -0,0 +1,167 @@
+ *
+ */
+
+#include "FreeRTOS.h"
+#include "task.h"
+#include <cstring>
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifndef NDEBUG
+#include <atomic>
+#include <cstdio>
+#endif
+
+#if CHIP_CONFIG_MEMORY_DEBUG_DMALLOC
+#include <dmalloc.h>
+#include <support/SafeInt.h>
+#endif // CHIP_CONFIG_MEMORY_DEBUG_DMALLOC
+
+/* Assumes 8bit bytes! */
+#define heapBITS_PER_BYTE ((size_t) 8)
+
+/* Define the linked list structure.  This is used to link free blocks in order
+of their memory address. */
+typedef struct A_BLOCK_LINK
+{
+    struct A_BLOCK_LINK * pxNextFreeBlock; /*<< The next free block in the list. */
+    size_t xBlockSize;                     /*<< The size of the free block. */
+} BlockLink_t;
+
+/* The size of the structure placed at the beginning of each allocated memory
+block must by correctly byte aligned. */
+static const size_t xHeapStructSize =
+    (sizeof(BlockLink_t) + ((size_t)(portBYTE_ALIGNMENT - 1))) & ~((size_t) portBYTE_ALIGNMENT_MASK);
+
+/* Gets set to the top bit of an size_t type.  When this bit in the xBlockSize
+member of an BlockLink_t structure is set then the block belongs to the
+application.  When the bit is free the block is still part of the free heap
+space. */
+static size_t xBlockAllocatedBit = ((size_t) 1) << ((sizeof(size_t) * heapBITS_PER_BYTE) - 1);
+
+extern "C" {
+
+size_t xPortMallocUsableSize(void * pv)
+{
+    uint8_t * puc = (uint8_t *) pv;
+    BlockLink_t * pxLink;
+    void * voidp;
+    size_t sz = 0;
+
+    if (pv != NULL)
+    {
+        vTaskSuspendAll();
+        {
+            /* The memory being checked will have an BlockLink_t structure immediately
+            before it. */
+            puc -= xHeapStructSize;
+
+            /* This casting is to keep the compiler from issuing warnings. */
+            voidp  = (void *) puc;
+            pxLink = (BlockLink_t *) voidp;
+
+            /* Check if the block is actually allocated. */
+            configASSERT((pxLink->xBlockSize & xBlockAllocatedBit) != 0);
+            configASSERT(pxLink->pxNextFreeBlock == NULL);
+
+            sz = (pxLink->xBlockSize & ~xBlockAllocatedBit) - xHeapStructSize;
+        }
+        (void) xTaskResumeAll();
+    }
+
+    return sz;
+}
+
+void * __wrap_malloc(size_t size)
+{
+    return pvPortMalloc(size);
+}
+
+void __wrap_free(void * ptr)
+{
+    vPortFree(ptr);
+}
+
+void * __wrap_calloc(size_t num, size_t size)
+{
+    size_t total_size = num * size;
+
+    if (size && total_size / size != num)
+        return nullptr;
+
+    void * ptr = pvPortMalloc(total_size);
+    if (ptr)
+        memset(ptr, 0, total_size);
+    return ptr;
+}
+
+void * __wrap_realloc(void * ptr, size_t new_size)
+{
+    if (new_size)
+    {
+        void * new_ptr = pvPortMalloc(new_size);
+        if (new_ptr)
+        {
+            if (ptr)
+            {
+                size_t old_ptr_size = xPortMallocUsableSize(ptr);
+                if (new_size >= old_ptr_size)
```

Comment:
```suggestion
                if (new_size > old_ptr_size)
```

If size is same as before, and you assume it's OK to keep if smaller, it's also OK to keep original if same size.

Reply:
Fixed